### PR TITLE
PL-Standings Implementation

### PR DESCRIPTION
**Pull request by T-Sajad**
### Changes

- Added another fecth request to get the pl standings data from the api endpoint
- Data from the fecth request is stored in a JSON file
- Created react component named standingsList.jsx
- Used the javascript map prototype to iterate over the  data in the JSON file and create new array of consisiting premier team names , points and goal difference
- Created a child react component named standings to contain the html and structure for  the premeir league table. This was possible by passing the standing prop down to the child component.

### Future updates

- Its possible to further clean the code and refactor
- Need to implement a fetaure where the premeir league table updates every game day

### Challenges

- Trying to access the arrays in the json respsonse.

image

image

image

image

image

Link to the issue and design below:
https://github.com/T-Sajad/football-scores-app/issues/2#issuecomment-1794929887

### CODE REVIEW

**Review by josephjclark — CHANGES_REQUESTED** [review]
Hi @T-Sajad 

So the body of the pull request here is the same as the issue?

This isn't right. You should have:
1) Updated the issue with the text in my comment (with any changes that you think might be necessary), so that the issue is clean, up to date, and accurate
2) Written a brief overview of the work you've done here.

You've even copied the issue title into the PR title - I can see because it's got the same typo in it :sweat_smile: 

Please take more time and consideration with this stuff. It's actually really important.

When creating a PR to review, you are asking someone to spend their time looking at your work. The job of a PR description is to make it easy for the reviewer, help them spend less time.

To steal an analogy from my boss: in football, the midfielder helps the striker to score by playing a controlled, weighted pass. You don't just hoof the ball upfield and expect Harry Kane to do all the work (as Spurs have done for years). You play the ball just ahead of his feet.

For example, you've asked me to review the code - but you haven't bothered to update the issue, write an overview of what you've done, or done anything to help my review easier. It looks like you've spent minimum effort on this, while I've spent maximum effort.

I mean, I've been here for ten minutes writing this message and I haven't even looked at your code yet.

You can use the PR body to highlight anything you struggled with, anything unexpected, anything you've done that's outside the scope of the issue, anything you haven't done, or just tell me where to start the review. Screenshots or even videos are also useful so that I don't have to download and run the code.

It's always a good idea to assume that whoever is reading the PR isn't intimately related with your work.

Oh, and you should also link to the issue related to the work. In github you can literally write #2  (literally `#2` ) and the issue will be nicely linked.

Here's an example of a PR I raised at work a little back:https://github.com/OpenFn/Lightning/pull/942 Your PR doesn't need to be anything like as long as this, but it just gives you an idea of the kind of thing I think about when raising a PR.

Please update the PR with a brief description of what you've done (and haven't done) and I'll take a closer look at the code.

**Comment by josephjclark** [comment]
Thank you @T-Sajad, this looks much better.

I'll have a look at the code in the morning, but first, an idea (based on your very helpful screenshot). How about putting the table on the left, and the tabs on the right ? A side-by-side layout, rather than top/bottom as you have now?

File: scripts/get-code.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import 'dotenv/config';
+import fs from 'node:fs';
+
+const candidateId = 374;
+const url = `https://apply.scorchsoft.dev/api/candidate/${candidateId}/get-code`;
+const options = {
+  method: 'POST',
+  body: JSON.stringify({
+    email: 'tsajad300@gmail.com',
```
Comment (josephjclark) [inline scripts/get-code.js:9]: You should always avoid checking credentials into github!

Usually you'd store these in environment variables and load them in here from `process.env`

```
email: process.env.FOOTBALL_EMAIL
password: process.env.FOOTBALL_PASSWORD

```
I think you have a `.env` file which can store the vrariables

File: scripts/get-standings.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import 'dotenv/config';
+import fs from 'node:fs';
+
+const url =
+  'https://api-football-beta.p.rapidapi.com/standings?season=2023&league=39';
+const options = {
+  method: 'GET',
+  headers: {
+    'X-RapidAPI-Key': process.env.API_KEY,
```
Comment (josephjclark) [inline scripts/get-standings.js:9]: This is an example of  good practice for handling credentials :+1:

File: src/components/Standings.jsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { tr } from 'date-fns/locale';
+import React from 'react';
+
+function Standings({ standing }) {
+  return (
+    <tr>
+      <td className="values">{`${standing.rank}`}</td>
```
Comment (josephjclark) [inline src/components/Standings.jsx:7]: You don't the embed the value in a string here - you can just  do this:
```
<td className="values">{standing.rank}</td>
```

File: src/components/StandingsList.jsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import standingData from '../../data/get-standings.json';
+import Standings from './Standings';
+
+function StandingsList() {
+  const standings = standingData.response[0].league.standings[0]; // Note the correct property name here
```
Comment (josephjclark) [inline src/components/StandingsList.jsx:6]: We don't have to address this here (maybe you can raise an issue for it and we'll do it later), but I want to respond to this because I think you mentioned it in your PR overview:

This very long chain of properties to get the data you need is ugly (it doesn't look nice) brittle (it can easily go wrong ) and unintuitive (it's impossible to guess and hard to lookup what this path might be).

Usually what we'd do to solve this is TRANSFORM the original response data we get from the server. We'd do this in our request script before we save it to JSON (or otherwise send it on to the app).

The transformer function needs to anticipate the data needed by the app - which in this case, is just the premier league table.

So basically what you do is:
* Fetch the data
* Pull out `response[0].league.standings[0]` from the response data into a variable
* Save the extracted data into your JSON file.

The transformer may also remove unused data or change property names.

Once your data has been nicely processed, it's much easier, more intuitive and more robust to use in the frontend.

**Review by josephjclark — CHANGES_REQUESTED** [review]
This is actually really good Talal - the code is nice and clean, and it JUST does what you've set out to do. I'm impressed.

I'm requesting one change around your credentials, and there's a couple of general comments you may want to address or think about (but it's fine to leave them for now).

As I said in the main thread - I do think this design would be visually more successful if the table and tabs were arranged in columns (side-by-side) rather than rows.

File: src/components/Standings.jsx (hunk 3 above)
Comment (T-Sajad) [reply src/components/Standings.jsx:7]: fixed thanks.
